ledger/participant-state/kvutils/tools/integrity-check/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/RawPreExecutingCommitStrategySupport.scala

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
```

Comment:
Can you add some tests for this, please? This has non-trivial logic e.g. around handling of record time.

Reply:
We have tests! :grin:

Reply:
In order to _easily_ write tests for this, I first had to extract out some unrelated logic from `IntegrityChecker` and the two `Support` classes which diffed write sets. I have moved this logic, and the associated tests, into a class named `WriteSetComparison`.